### ISSUE TITLE
`lines-around-comment`, allowBlockStart, and ES2015 Classes. 

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#2894]
Currently, even if `allowBlockStart` is `true`, this rule warns a comment that is at the top in a class body.
This rule should not warn it.

----

```js
class A {
    /**
     * test
     */
    constructor() {
        /*
         * test
         */
    }
}
```

```
> eslint -v
v0.24.0
> eslint test.js --reset --no-eslintrc --env es6 --rule "lines-around-comment:[2,{allowBlockStart:true}]"
```

## Actual

```
test.js
  2:4  error  Expected line before comment  lines-around-comment

✖ 1 problem (1 error, 0 warnings)
```

## Expected

No errors.

**Comment by gyandeeps** [comment]
Yup, I saw the code and we are not accounting for class syntax for the above scenario.

**Comment by mysticatea** [comment]
Thank you!
